src/edu/washington/escience/myria/operator/CSVFileScanFragment.java

```diff
@@ -430,6 +441,23 @@ protected void init(final ImmutableMap<String, Object> execEnvVars) throws DbExc
                 new BufferedReader(new InputStreamReader(partitionInputStream)),
                 CSVFormat.newFormat(delimiter).withQuote(quote).withEscape(escape));
         iterator = parser.iterator();
+
+        try {
+          if (!iterator.hasNext()) {
+            flagAsIncomplete = true;
+          }
+        } catch (Exception e) {
+          /*
+           * FIX ME: If we hit an exception for a malformed row (in case of quotes for example), we mark this as the
+           * last row
+           */
+          if (e.getMessage() != null && e.getMessage().contains(truncatedQuoteErrorMessage)) {
+            onLastRow = true;
+          } else {
+            throw e;
+          }
+        }
+
         for (int i = 0; i < numberOfSkippedLines; i++) {
```

Comment:
So I think I'll only mention it on a comment for now. I don't immediately see how we can handle the case where other workers (besides the first one) have skipped lines. A worker would only know which lines to skip given the number of lines the previous workers have skipped. This sounds like it would have to be some sort of post-ingest-processing step that would might require worker communication. I don't think we'll be able to do this in parallel if we wanted to handle it.